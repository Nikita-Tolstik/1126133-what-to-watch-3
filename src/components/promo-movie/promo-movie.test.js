import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import PromoMovie from './promo-movie.jsx';
import history from '../../history.js';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const mockFilms = [
  {
    id: 1,
    title: `1`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Thriller`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 2,
    title: `2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Thriller`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 3,
    title: `3`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Thriller`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 4,
    title: `4`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Thriller`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 5,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 6,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 7,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 8,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 9,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 10,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 11,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
  {
    id: 12,
    title: `Fantastic Beasts: The Crimes of Grindelwald2`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    description: `Fantastic Beasts: The Crimes of Grindelwald`,
    rating: 5,
    scoresCount: 7,
    director: `Wes Andreson`,
    stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: 55,
    genre: `Action`,
    released: 44,
    isFavorite: true,
    videoLink: `https://upload.wikimedia.org/`,
    videoPreview: `https://upload.wikimedia.org/`,
  },
];

const mockFilm = {
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald2`,
  posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  description: `Fantastic Beasts: The Crimes of Grindelwald`,
  rating: 5,
  scoresCount: 7,
  director: `Wes Andreson`,
  stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
  runTime: 55,
  genre: `Thriller`,
  released: 44,
  isFavorite: true,
  videoLink: `https://upload.wikimedia.org/`,
  videoPreview: `https://upload.wikimedia.org/`,
};


it(`Render PromoMovie - authorizationStatus = AUTH`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      favoriteFilms: mockFilms,
      promoFilm: mockFilm,
      isError: false,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      userInfo: `xxxxxxxxxxxxxxxxxxxxxxxxxxxxx`,
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <PromoMovie
            onCardFilmClick={() => {}}
          >
          </PromoMovie>
        </Provider>
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();


  expect(tree).toMatchSnapshot();

});

it(`Render PromoMovie - authorizationStatus = NO_AUTH`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      favoriteFilms: [],
      promoFilm: mockFilm,
      isError: true,
    },
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`,
      userInfo: `no info`,
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <PromoMovie
            onCardFilmClick={() => {}}
          >
          </PromoMovie>
        </Provider>
      </Router>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();


  expect(tree).toMatchSnapshot();

});

