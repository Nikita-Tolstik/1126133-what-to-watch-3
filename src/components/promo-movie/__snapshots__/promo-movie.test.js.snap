// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PromoMovie - authorizationStatus = AUTH 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald2"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="https://htmlacademy-react-3.appspot.comxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald2"
          height="327"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Fantastic Beasts: The Crimes of Grindelwald2
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Thriller
          </span>
          <span
            className="movie-card__year"
          >
            44
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="14"
              viewBox="0 0 18 14"
              width="18"
            >
              <use
                xlinkHref="#in-list"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Render PromoMovie - authorizationStatus = NO_AUTH 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald2"
      src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald2"
          height="327"
          src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Fantastic Beasts: The Crimes of Grindelwald2
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Thriller
          </span>
          <span
            className="movie-card__year"
          >
            44
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
